reactor-netty-core/src/main/java/reactor/netty/transport/TransportConnector.java

```diff
@@ -102,8 +105,24 @@
 		Objects.requireNonNull(resolverGroup, "resolverGroup");
 		Objects.requireNonNull(channelInitializer, "channelInitializer");
 
-		return doInitAndRegister(config, channelInitializer, remoteAddress instanceof DomainSocketAddress)
-				.flatMap(channel -> doResolveAndConnect(channel, config, remoteAddress, resolverGroup));
+		boolean isDomainAddress = remoteAddress instanceof DomainSocketAddress;
+		return doInitAndRegister(config, channelInitializer, isDomainAddress)
+				.flatMap(channel -> doResolveAndConnect(channel, config, remoteAddress, resolverGroup)
+						.onErrorResume(RetryConnectException.class,
+								t -> {
+									AtomicInteger index = new AtomicInteger(1);
+									return Mono.<Channel>create(sink ->
```

Comment:
why use a `Mono.create` here when `doInitAndRegister` already returns  a `Mono`?

Reply:
I need `retryWhen` to execute again `doInitAndRegister`, if I have the code below, it retries only the `doConnect` functionality.

```
return doInitAndRegister(config, channelInitializer, isDomainAddress)
		.flatMap(ch -> {
			MonoChannelPromise mono = new MonoChannelPromise(ch);
			doConnect(t.addresses, config.bindAddress(), mono, index.get());
			return mono;
		})
		.retryWhen(Retry.max(t.addresses.size() - 1)
						.filter(RETRY_PREDICATE)
						.doBeforeRetry(sig -> index.incrementAndGet()));
```

Reply:
using a `defer` would be an adequate yet better solution I think.

Reply:
thanks, changing ...